xarray/tests/test_cftime_offsets.py

```diff
@@ -0,0 +1,723 @@
+import pytest
+
+from itertools import product
+
+import numpy as np
+import pandas as pd
+
+from xarray.coding.cftime_offsets import (
+    BaseCFTimeOffset, YearBegin, YearEnd, MonthBegin, MonthEnd,
+    Day, Hour, Minute, Second, _days_in_month,
+    to_offset, get_date_type, _MONTH_ABBREVIATIONS, _cftime_range,
+    to_cftime_datetime, cftime_range)
+from xarray import CFTimeIndex
+from . import has_cftime
```

Comment:
Instead of checking `has_cftime` and `skipif` decorators on every function, use `cftime = pytest.importorskip('cftime')`.